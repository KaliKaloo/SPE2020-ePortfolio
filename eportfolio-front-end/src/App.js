import React, {useEffect, useState} from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
import Landing from './pages/Landing';
import Login from './pages/Login';
import Register from './pages/Register';
import TermsAndConditions from './pages/TermsAndConditions.js';

import Home from './pages/categories/Home';
import Projects from './pages/categories/Projects';
import WorkExpr from './pages/categories/WorkExpr';
import Qualification from './pages/categories/Qualification';
import Achievements from './pages/categories/Achievements';
import EditorPage from './pages/EditorPage';

import SettingsGeneral from './pages/settings/setting_general';
import SettingsPrivacy from './pages/settings/setting_privacy';
import SettingsAddRess from './pages/settings/setting_addRes';


import Publish from './published/Publish'


function App() {
  document.title = `ePortfolio`;

  const [isLog, setisLog] = useState(false);

  useEffect(() => {
    let isCancelled = false;
    if (!isCancelled) {
      if (localStorage.getItem('user') !== null) {
        setisLog(true);
      }
    } else {
      setisLog(false)
    }
    return () => {
      isCancelled = true
    };
  }, [isLog]);

  return (
    <>
      <Router>
        <Switch>
        
          <Route exact path="/login" component={Login}/>
          <Route exact path="/register" component={Register}/>
          <Route path='/terms_and_conditions' component={TermsAndConditions}/>

          <Route path='/projects' component={Projects}/>
          <Route path='/workExperience' component={WorkExpr}/>
          <Route path='/achievements' component={Achievements}/>
          <Route path='/qualification' component={Qualification}/>

          <Route path='/editor/:postid?' component={EditorPage}/>
          
          <Route path='/settings_general' component={SettingsGeneral}/>
          <Route path='/settings_privacy' component={SettingsPrivacy}/>
          <Route path='/settings_additional_res' component={SettingsAddRess}/>

          <Route path={['/publish/home/:username?', '/publish/:username?']} component={Publish}/>

          <Route path='/Home' component={Home}/>

          <Route path={['/']}>
            {
              isLog ?
                <Home/>
                :
                <Landing/>
            }
          </Route>
        </Switch>
      </Router>
    </>
  );
}

export default App;