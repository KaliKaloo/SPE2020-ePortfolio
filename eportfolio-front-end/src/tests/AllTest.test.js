import React from 'react'
import { render,cleanup } from '@testing-library/react'
import App from '../App.js'

afterEach(cleanup)

it('should take a snapshot for the whole App UI', () => {
  const { asFragment } = render(<App />)

  expect(asFragment(<App />)).toMatchSnapshot()
})


