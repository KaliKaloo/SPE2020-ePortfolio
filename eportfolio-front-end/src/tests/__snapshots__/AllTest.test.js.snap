// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot for the whole App UI 1`] = `
<DocumentFragment>
  <div
    class="home__hero-section darkBg"
  >
    <div
      class="container"
    >
      <div
        class="row home__hero-row"
        style="display: flex; flex-direction: row;"
      >
        <div
          class="col"
        >
          <div
            class="home__hero-text-wrapper"
          >
            <div
              class="top-line"
            >
              Overview
            </div>
            <div
              class="heading"
            >
              ePortfolio
            </div>
            <p
              class="home__hero-subtitle"
            >
              Create a portfolio to showcase yourself and your work. Ideal for those students who want to catalogue their progress
            </p>
            <a
              href="/register"
            >
              <button
                class="btn btn--primary btn--wide blue"
              >
                Get Started
              </button>
            </a>
            <a
              class="login_text light"
              href="/login"
            >
               Log in 
            </a>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="home__hero-img-wrapper"
          >
            <img
              alt="Credit Card"
              class="home__hero-img"
              src="images/svg-6.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="home__hero-section"
  >
    <div
      class="container"
    >
      <div
        class="row home__hero-row"
        style="display: flex; flex-direction: row-reverse;"
      >
        <div
          class="col"
        >
          <div
            class="home__hero-text-wrapper"
          >
            <div
              class="top-line"
            >
              EASY TO SETUP
            </div>
            <div
              class="heading dark"
            >
              Simple to get started
            </div>
            <p
              class="home__hero-subtitle dark"
            >
              Get everything set up and ready in under 5 minutes. All you need to do is add your information and you're ready to go.
            </p>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="home__hero-img-wrapper"
          >
            <img
              alt="Vault"
              class="home__hero-img"
              src="images/info.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="home__hero-section darkBg"
  >
    <div
      class="container"
    >
      <div
        class="row home__hero-row"
        style="display: flex; flex-direction: row;"
      >
        <div
          class="col"
        >
          <div
            class="home__hero-text-wrapper"
          >
            <div
              class="top-line"
            >
              Easy to use
            </div>
            <div
              class="heading"
            >
              Showcase your work in minutes 
            </div>
            <p
              class="home__hero-subtitle"
            >
              Effortlessly add as many of your projects and work experiences to your portfolio. Also display your images or attach files ito each post.
            </p>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="home__hero-img-wrapper"
          >
            <img
              alt="Vault"
              class="home__hero-img"
              src="images/post.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="home__hero-section"
  >
    <div
      class="container"
    >
      <div
        class="row home__hero-row"
        style="display: flex; flex-direction: row-reverse;"
      >
        <div
          class="col"
        >
          <div
            class="home__hero-text-wrapper"
          >
            <div
              class="top-line"
            >
              Easy to share
            </div>
            <div
              class="heading dark"
            >
              Share your portfolio
            </div>
            <p
              class="home__hero-subtitle dark"
            >
              Easily share your portfolio to anyone with just a simple link. 
            </p>
            <a
              href="/register"
            >
              <button
                class="btn btn--primary btn--wide blue"
              >
                Sign Up Now
              </button>
            </a>
            <a
              class="login_text dark"
              href="/login"
            >
               Log in 
            </a>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="home__hero-img-wrapper"
          >
            <img
              alt="Vault"
              class="home__hero-img"
              src="images/share.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
